.travis.yml

```diff
@@ -31,26 +16,74 @@ stages:
   - build
 jobs:
   include:
-    - &test_unit
+    - name: Unit tests
       stage: test
-      script: yarn lint && yarn test
-      name: Unit tests
-      os: linux
-    - &build_web
+      <<: *node
+      script:
+        - yarn lint
+        - yarn test
+    - name: Web build
       stage: build
-      script: yarn build
-      name: Web build
-      os: linux
+      <<: *node
+      script:
+        - yarn build
+      before_deploy:
+        - yarn secrets:decrypt
+        - eval "$(ssh-agent -s)"
+        - chmod 600 ./scripts/decrypted/id_rsa
+        - ssh-add ./scripts/decrypted/id_rsa
       deploy:
         - provider: script
           repo: cozy/cozy-banks
           skip-cleanup: true
-          script: yarn cozyPublish && yarn uploadStaticFiles
+          script:
+            - yarn cozyPublish
+            - yarn uploadStaticFiles
           on:
             branch: master
         - provider: script
           repo: cozy/cozy-banks
           skip-cleanup: true
-          script: yarn cozyPublish && yarn uploadStaticFiles
+          script:
+            - yarn cozyPublish
+            - yarn uploadStaticFiles
           on:
             tags: true
+    - name: Android build
+      stage: build
+      os: linux
+      language: android
+      android:
+        components:
+          - build-tools-26.0.2
+      before_script:
+        - yes | sdkmanager "platforms;android-27" # see https://github.com/travis-ci/travis-ci/issues/8874#issuecomment-350350607
```

Comment:
I think we need a set -e to fail on errors. I am not sure that Travis does that automatically.

Reply:
What is the difference between before_script and script ? Why not put everything in a single `scripts/build-android.sh`. I am not a fan of having long scripts inside `.travis.yml`.